ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -768,9 +767,9 @@ public class RNTrackPlayer: RCTEventEmitter, AudioSessionControllerDelegate {
         }
     }
 
-    private func getPlaybackStateBodyKeyValues() -> Dictionary<String, Any> {
-        var body: Dictionary<String, Any> = ["state": State.fromPlayerState(state: player.playerState).rawValue]
-        if (player.playerState == AudioPlayerState.failed) {
+    private func getPlaybackStateBodyKeyValues(state: AudioPlayerState) -> Dictionary<String, Any> {
+        var body: Dictionary<String, Any> = ["state": State.fromPlayerState(state: state).rawValue]
+        if (state == AudioPlayerState.failed) {
             body["error"] = getPlaybackStateErrorKeyValues()
```

Comment:
Just curious and want to learn more about the codebase; shouldn't the state be passed to the `getPlaybackStateErrorKeyValues` as well, if changing this? Now The `state === AudioPlayerState.failed` check will not be the same state as the error-values function are checking on in line 741?